### PR TITLE
[RLLib] Make DQN update_target use only trainable variables.

### PR DESCRIPTION
**Pull request by n30111**
## Why are these changes needed?

This PR makes changes in the DQN `update_target` to use only trainable variables. This makes the behavior comparable with DDPG implementation https://github.com/ray-project/ray/blob/master/rllib/algorithms/ddpg/ddpg_tf_policy.py#L433. 

This is needed for situations when we want to use non-trainable variables in the base model but not in the target model for various logistic reasons. 

These changes won't affect the performance or correctness. 
## Related issue number

N/A

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by gjoliver** [comment]
this seems like a reasonable change. but
1. we should stop creating all the policy.target_q_func_vars and policy.q_func_vars whenever this TargetNetworkMixin is used.
2. I think this may have performance implication, now that we are getting the trainable variable multiple times.
Another way of doing this is to initialize policy.target_q_func_vars to only trainable variables.

**Comment by gjoliver** [comment]
oh and also, we should actually create these variables inside TargetNetworkMixIns constructor, so these things are self-contained.

**Comment by n30111** [comment]
@gjoliver thanks, I have moved the vars out of the method to the constructor to avoid repetitive copy. I did not change the others properties defined (as these were used `self.variables`).

File: rllib/policy/tf_mixins.py [hunk 1]
```diff
@@ -290,16 +290,19 @@ def __init__(
         action_space: gym.spaces.Space,
         config: TrainerConfigDict,
     ):
+        q_func_vars = self.model.trainable_variables()
+        target_q_func_vars = self.target_model.trainable_variables()
```
Comment (gjoliver) [inline rllib/policy/tf_mixins.py:294]: sorry about the delay.
I took another careful look at this, and I feel like we actually don't have to change anything in this __init__()?
what is necessary is to simply change the q_func_vars(self) function and target_q_func_vars(self) function below to call trainable_variables() right?

sorry I didn't give the best suggestion in my first review.

also, what do you think we should do for the torch version of TargetNetworkMixin?

Thanks.

Comment (n30111) [reply rllib/policy/tf_mixins.py:294]: @gjoliver I thought some other part of the code may be using the `policy.variables`, so did not touch that part. If no other code uses that, we can make the change in the respective methods, I'm not 100% sure about, can you please confirm that.

It seems for both DDPG and DQN, torch policy uses the same logic, in both of the cases it updates the entire `state_dict`.

Comment (gjoliver) [reply rllib/policy/tf_mixins.py:294]: good question. our code really is a mess. we should have implemented get_states() and set_states() for the TargetNetworkMixIn instead, sigh.

can you do me a small favor then, just rename these 2 variables to:
trainable_q_func_vars and trainable_target_q_func_vars please.

Comment (n30111) [reply rllib/policy/tf_mixins.py]: Thanks, updated the variables names.

**Review by gjoliver — APPROVED** [review]
one 1 lint. any chance you can run ./ci/lint/format.sh, rebase to master, and send another patch?
I will ping @sven1977 to merge this.

**Comment by n30111** [comment]
@gjoliver I have rebased the code and ran format.sh, please check.

**Pull request merged**